### PR TITLE
BYOVPC for AWS

### PR DESCRIPTION
**Pull request by micheleRP**
## Description

Resolves https://github.com/redpanda-data/documentation-private/issues/2359
Review deadline: Aug 6

Status: This is in private beta.

Because Cloud docs moved into a separate repo, I had to move https://github.com/redpanda-data/docs/pull/550 into this new PR.

This also nests AWS, Azure, and GCP folders under Create BYOC in the side nav (+ adds aliases and updates links to the new file locations).

## Page previews

- [What's New](https://deploy-preview-12--rp-cloud.netlify.app/redpanda-cloud/get-started/whats-new-cloud/)
- [Configure a Customer-Managed VPC on AWS](https://deploy-preview-12--rp-cloud.netlify.app/redpanda-cloud/get-started/cluster-types/byoc/vpc-byo-aws/)
- [Cloud Overview](https://deploy-preview-12--rp-cloud.netlify.app/redpanda-cloud/get-started/cloud-overview/#cluster-types)

## Checks

- [X] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

File: modules/get-started/pages/cluster-types/byoc/vpc-byo-aws.adoc [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+  description = "Minimum permissions required for rpk user for BYO VPC"
+  policy      = each.value.json
+}
+```
+====
+
+== Limitations
+
+* Existing clusters cannot be moved to a customer-managed VPC.
+* After creating a cluster with a customer-managed VPC, you cannot change to a different VPC.
```
Comment (Feediver1) [inline modules/get-started/pages/cluster-types/byoc/vpc-byo-aws.adoc:341]:
```suggestion
* You cannot move existing clusters to a customer-managed VPC.
* You cannot change to a different VPC after creating a cluster with a customer-managed VPC.
```

File: modules/get-started/pages/cluster-types/byoc/vpc-byo-aws.adoc [hunk 2]
```diff
@@ -0,0 +1,593 @@
...
+
+== Configure your VPC
+
+Terraform modules contain the specifications for the required resources. Terraform creates the following resources and outputs ARN values necessary in later steps.
```
Comment (Feediver1) [inline modules/get-started/pages/cluster-types/byoc/vpc-byo-aws.adoc:345]:
```suggestion
Terraform modules contain the specifications for the required resources. Terraform creates the following resources and outputs ARN values necessary in later steps:
```

File: modules/get-started/pages/whats-new-cloud.adoc [hunk 3]
```diff
@@ -6,6 +6,12 @@
 
 This page lists new features added in Redpanda Cloud.
 
+== August 2024
+
+=== Customer-managed VPC on AWS
+
+With standard BYOC clusters, Redpanda manages security policies and resources for your VPC, including subnetworks, service accounts, IAM roles, firewall rules, and storage buckets. For the most security, you can manage these resources yourself with a customer-managed VPC. Repdanda now provides a xref:get-started:cluster-types/byoc/vpc-byo-aws.adoc[customer-managed VPC on AWS] offering, in addition to xref:get-started:cluster-types/byoc/vpc-byo-gcp.adoc[customer-managed VPC on GCP].
```
Comment (Feediver1) [inline modules/get-started/pages/whats-new-cloud.adoc:13]:
```suggestion
With standard BYOC clusters, Redpanda manages security policies and resources for your VPC, including subnetworks, service accounts, IAM roles, firewall rules, and storage buckets. For the highest level of security, you can manage these resources yourself with a customer-managed VPC. Repdanda now provides a xref:get-started:cluster-types/byoc/vpc-byo-aws.adoc[customer-managed VPC on AWS] offering, in addition to xref:get-started:cluster-types/byoc/vpc-byo-gcp.adoc[customer-managed VPC on GCP].
```

File: modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-aws.adoc [hunk 4]
```diff
@@ -4,6 +4,8 @@
 
 To create a Redpanda cluster in your virtual private cloud (VPC), log in to https://cloud.redpanda.com[Redpanda Cloud^]. 
 
+NOTE: With standard BYOC clusters, Redpanda manages security policies and resources for your VPC, including subnetworks, service accounts, IAM roles, firewall rules, and storage buckets. For the most security, you can manage these resources yourself with a xref:./vpc-byo-aws.adoc[customer-managed VPC on AWS].
```
Comment (Feediver1) [inline modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-aws.adoc:7]:
```suggestion
NOTE: With standard BYOC clusters, Redpanda manages security policies and resources for your VPC, including subnetworks, service accounts, IAM roles, firewall rules, and storage buckets. For the highest level of security, you can manage these resources yourself with a xref:./vpc-byo-aws.adoc[customer-managed VPC on AWS].
```

**Comment by micheleRP** [comment]
Status 8/24/24: This should stay in private beta until we get feedback from another customer, per Praseed.

File: modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc [hunk 5]
```diff
@@ -0,0 +1,259 @@
+= Configure a Customer-Managed VPC on AWS
+:description: Connect Redpanda Cloud to your existing VPC for additional security.
+:page-aliases: get-started:cluster-types/byoc/vpc-byo-aws.adoc
+:page-cloud: true
+
+include::shared:partial$feature-flag.adoc[]
+
+With a standard BYOC cluster, Redpanda manages the VPC lifecycle. For additional security, you can deploy the Redpanda data plane into your existing shared VPC and manage the VPC lifecycle yourself. When you create a BYOC cluster, you specify your VPC and service account. The Redpanda Cloud agent doesn't create any new resources or alter any settings in your account. With a *customer-managed* VPC:
```
Comment (Feediver1) [inline modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc:8]: please link "data plane"  to the glossary defn here

File: modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+
+* You provide your own VPC in your AWS account.
+* You maintain more control of your  account, because Redpanda requires fewer permissions than standard BYOC clusters.
```
Comment (Feediver1) [inline modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc:11]:
```suggestion
* You maintain more control over your account, because Redpanda requires fewer permissions than standard BYOC clusters.
```

File: modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+* You control your security resources and policies, including subnets, service accounts, IAM roles, firewall rules, and storage buckets.
+
+== Prerequisites
+
+* Familiarity with the xref:api:ROOT:cloud-api.adoc[Redpanda Cloud API]. For example, you use the Cloud API to authenticate and create a cluster.
```
Comment (Feediver1) [inline modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc:16]:
```suggestion
* Familiarity with the xref:api:ROOT:cloud-api.adoc[Redpanda Cloud API]. For example, you should familiarize yourself with how to use the Cloud API to authenticate and create a cluster.
```

File: modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+* Access to an AWS project in which you create your cluster.
+* Minimum permissions in that AWS project. For the actions required by the user who will create the cluster with `rpk cloud byoc aws apply`, see https://github.com/redpanda-data/cloud-examples/blob/main/customer-managed/aws/terraform/iam_rpk_user.tf[`iam_rpk_user.tf`^].
+
+== Limitations
+
+* You cannot move existing clusters to a customer-managed VPC.
+* You cannot change to a different VPC after creating a cluster with a customer-managed VPC.
```
Comment (Feediver1) [inline modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc:23]: So does this mean you can have one and only one customer-managed VPC amongst multiple clusters?

File: modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+
+== Create managed resources
+
+For information about the Terraform code containing the specifications for the required resources, see the https://github.com/redpanda-data/cloud-examples/tree/main/customer-managed/aws[customer-managed VPC on AWS README^].
+The ARNs output by this Terraform code are necessary in later steps.
```
Comment (Feediver1) [inline modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc:28]:
```suggestion
The ARNs (Amazon Resource Names) output by this Terraform code are necessary in later steps.
```

File: modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+https://redpanda-api-documentation.netlify.app/#post-/v1beta2/networks[Create Network].
+
+```
+curl -X POST "https://api.redpanda.com/v1beta2/networks" \
+ -H "accept: application/json"\
+ -H "content-type: application/json" \
+ -H "authorization: Bearer $YOUR_TOKEN" \
+-d @- << EOF
+{
+	 "name":"<name you choose for your network resource>",
```
Comment (Feediver1) [inline modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc:44]: maybe more simply: <your network resource name>
<network resource group id>

File: modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc [hunk 11]
```diff
@@ -0,0 +1,259 @@
...
+   "operation":{
+      "id":"cpas8k6r4up5li18auh0",
+      "metadata":{
+         "@type":"type.googleapis.com/redpanda.api.controlplane.v1beta2.CreateNetworkMetadata",
+         "network_id":"cpb338gekjj5i1cpj3t0"
+      },
+      "state":"STATE_IN_PROGRESS",
+      "started_at":"2024-05-28T19:33:54.631Z",
+      "type":"TYPE_CREATE_NETWORK",
+      "resource_id":"cpb338gekjj5i1cpj3t0"
```
Comment (Feediver1) [inline modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc:82]: maybe highlight this line?

File: modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+      },
+      "k8s_cluster_role": {
+        "arn": "<k8s_cluster_role_arn from terraform outputs>"
+      },
+      "cloud_storage_bucket": {
+        "arn": "<cloud_storage_bucket_arn from terraform outputs>"
+      }
+    }
+  },
+  # <This aws_private_link section is optional. See the https://docs.redpanda.com/current/deploy/deployment-option/cloud/aws-privatelink/[AWS documentation] for more information>
```
Comment (Feediver1) [inline modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc:154]: did you mean for the cross ref to include tagging here in this comment? If one was to copy and paste the entire string https://docs.redpanda.com/current/deploy/deployment-option/cloud/aws-privatelink/[AWS documentation] they would get a page not found result.

**Review by Feediver1 — APPROVED** [review]
See comments/suggestions

File: modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc (hunk 12 above)
Comment (micheleRP) [reply modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc:154]: thanks, I've fixed this!

File: modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc (hunk 11 above)
Comment (micheleRP) [reply modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc:82]: done

File: modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc [hunk 13]
```diff
@@ -0,0 +1,260 @@
...
+* Minimum permissions in that AWS project. For the actions required by the user who will create the cluster with `rpk cloud byoc aws apply`, see https://github.com/redpanda-data/cloud-examples/blob/main/customer-managed/aws/terraform/iam_rpk_user.tf[`iam_rpk_user.tf`^].
+
+== Limitations
+
+* You cannot move existing clusters to a customer-managed VPC.
+* You cannot change to a different VPC after creating a cluster with a customer-managed VPC.
+
+== Create managed resources
+
+For information about the Terraform code containing the specifications for the required resources, see the https://github.com/redpanda-data/cloud-examples/tree/main/customer-managed/aws[customer-managed VPC on AWS README^].
```
Comment (bpraseed) [inline modules/get-started/pages/cluster-types/byoc/AWS/vpc-byo-aws.adoc:27]: I would reword this to imply this is an example blueprint/template of TF code for the resources that's expected to be provisioned by the customer. So to avoid confusion this is mandatory to use and not a product feature.

File: modules/get-started/pages/cluster-types/byoc/aws/vpc-byo-aws.adoc [hunk 14]
```diff
@@ -0,0 +1,260 @@
+= Configure a Customer-Managed VPC on AWS
+:description: Connect Redpanda Cloud to your existing VPC for additional security.
+:page-aliases: get-started:cluster-types/byoc/vpc-byo-aws.adoc
+:page-cloud: true
+
+include::shared:partial$feature-flag.adoc[]
+
+With a standard BYOC cluster, Redpanda manages the VPC lifecycle. For additional security, you can deploy the Redpanda glossterm:data plane[] into your existing shared VPC and manage the VPC lifecycle yourself. When you create a BYOC cluster, you specify your VPC and service account. The Redpanda Cloud agent doesn't create any new resources or alter any settings in your account. With a *customer-managed* VPC:
```
Comment (frenchfrywpepper) [inline modules/get-started/pages/cluster-types/byoc/aws/vpc-byo-aws.adoc:8]: > The Redpanda Cloud agent doesn't create any new resources or alter any settings in your account.

This feels a little off. We will create many resources, but we will not create VPC or network resources.

**Review by frenchfrywpepper — APPROVED** [review]

**Comment by david-yu** [comment]
Providing gist [here](https://gist.github.com/david-yu/f128bd3b87f71d7268348553d6421c34) of my working updates to commands in this doc. High level comments. 

1. We should make it very easy to copy and paste commands so little surgery is required to input the right values into json files. Allow users to input values via env variables and re-use them across inputs. 
2. AWS and Azure instructions should be similar in structure. We should also include commands like check cluster status on Azure. 
3. We should also include teardown instructions for both AWS and Azure.

**Comment by JakeSCahill** [comment]
> Providing gist [here](https://gist.github.com/david-yu/f128bd3b87f71d7268348553d6421c34) of my working updates to commands in this doc. High level comments.
> 
> 1. We should make it very easy to copy and paste commands so little surgery is required to input the right values into json files. Allow users to input values via env variables and re-use them across inputs.
> 2. AWS and Azure instructions should be similar in structure. We should also include commands like check cluster status on Azure.
> 3. We should also include teardown instructions for both AWS and Azure.

Going to add this as a follow-up improvement because this one has been open for so long and we need it for the 24.3 announcement tomorrow.
